js/sales.js

```diff
@@ -123,6 +123,65 @@
         return dataCell;
     }
 
+    function getNewStore(event){
+        event.preventDefault();
+        const form = event.target;
+        const storeName = form.querySelector('#store-name').value;
+        const minHourlyCustomers = parseInt(form.querySelector('#min-hourly-customer').value);
+        const maxHourlyCustomers = parseInt(form.querySelector('#max-hourly-customer').value);
+        const avgCookiesPerCustomer = parseInt(form.querySelector('#avg-cookies-per-customer').value);
+        const newStore = new CookieStore(storeName, minHourlyCustomers, maxHourlyCustomers, avgCookiesPerCustomer);
+        cookieStoreLocations.push(newStore);
+        addNewStore(newStore);
+        form.querySelector('input[type=\'submit\']').disabled = true;
+        Array.from(form.querySelectorAll('input')).forEach(input => {
+            if(input.value !== 'Enter store'){
+                input.value = '';
+            }
+        });
+    }
+
+    function toggleTotalRowHighlight(){
+        if(cookieStoreLocations.length % 2 === 0){
+            document.querySelector('tfoot').removeAttribute('class', 'highlight');
+        } else {
+            document.querySelector('tfoot').setAttribute('class', 'highlight');
+        }
+    }
+
+    function newStoreNameInputValidation(event){
+        const storeNameInput = event.target;
+        if(!isNaN(storeNameInput.value)){
+            storeNameInput.value = '';
+            storeNameInput.setAttribute('class', 'name-warning-input');
+            document.querySelector('#hide-name-warning-message').setAttribute('id', 'name-warning-message');
+        } else {
+            storeNameInput.removeAttribute('class', 'name-warning-input');
+            document.querySelector('p').setAttribute('id', 'hide-name-warning-message');
+        }
+    }
+
+    function allInputsCompleted(event){
+        const form = event.target.parentNode;
+        const formInputs = Array.from(form.querySelectorAll('input'));
+        if(formInputs.find(input => input.value === '') === undefined){
+            document.querySelector('input[type=\'submit\']').disabled = false;
+        }
+    }
+
+    function addNewStore(newStore){
+        newStore.render();
+        const storeTotal = newStore.calculateStoreTotal();
+        document.querySelector('tbody').lastChild.appendChild(createDataCell(storeTotal, 'td'));
```

Comment:
Good use of the `lastChild` property here, I didn't know that was available. It's probably more efficient than querying `tr:last-child`.